### PR TITLE
feat: order details redesign

### PR DESCRIPTION
**Pull request by muller39**
Description:

https://virtocommerce.atlassian.net/browse/ST-3759
--

QA-test:

Demo-test:

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.19.0-pr-434-f868-f86801e9.zip

### CODE REVIEW

File: client-app/pages/account/order-details.vue [hunk 1]
```diff
@@ -1,93 +1,152 @@
...
         </div>
+      </div>
 
-        <!-- Gifts section -->
-        <AcceptedGifts :items="giftItems" />
+      <!-- Gifts section -->
+      <AcceptedGifts :items="giftItems" />
+
+      <!-- Order comment section -->
+      <VcSection v-if="order?.comment && !isMobile" class="shadow-inner mt-5 p-6 lg:shadow" content-classes="pt-4">
```
Comment (goldenmaya) [inline client-app/pages/account/order-details.vue:78]: use `<VcSectionWidget />` component.
VcSection will be removed from project soon.

File: client-app/pages/account/order-details.vue [hunk 2]
```diff
@@ -1,93 +1,152 @@
...
-          <template #title>
-            <h3 class="text-24 font-bold uppercase mb-4">
-              {{ $t("pages.account.order_details.order_comment_section.title") }}
-            </h3>
-          </template>
-          <p class="break-words text-15" v-for="line in order?.comment?.split('\n')" :key="line">{{ line }}</p>
-        </VcSection>
+        <!-- Billing Address Widget -->
+        <VcCardWidget :title="$t('pages.account.order_details.billing_address_card.title')" icon="truck">
+          <div class="flex flex-col space-y-1.5 text-15">
```
Comment (goldenmaya) [inline client-app/pages/account/order-details.vue:108]: use `<VcAddressInfo />` component.

File: client-app/pages/account/order-details.vue [hunk 3]
```diff
@@ -1,93 +1,152 @@
...
-                <Vendor v-if="group.vendor" :vendor="group.vendor" class="inline-flex flex-row items-end gap-x-3" />
-                <span v-else class="text-gray-400" v-t="`pages.account.order_details.empty_vendor_label`" />
-              </div>
-
-              <OrderLineItems :items="group.items" class="lg:rounded" />
-            </div>
-          </template>
+    <VcLayoutWithRightSidebar is-sidebar-sticky>
+      <!-- Order Data Widget -->
+      <VcCardWidget v-if="isMobile" :title="$t('pages.account.order_details.order_data_card.title')" hideMobileTitle>
```
Comment (goldenmaya) [inline client-app/pages/account/order-details.vue:32]: use VcSectionWidget
we use VcCardWidget for sidebar only

File: client-app/pages/account/order-details.vue [hunk 4]
```diff
@@ -1,93 +1,152 @@
...
         </div>
-
-        <!-- Items not grouped by Vendor -->
-        <div v-else class="bg-white lg:mb-6 lg:rounded -mx-5 md:mx-0 lg:shadow-md-x lg:pt-5 lg:px-7">
-          <div class="bg-white shadow-light-lg mb-4 p-7 lg:mb-0 lg:pb-5 lg:px-0 lg:pt-0 lg:rounded lg:shadow-none">
-            <OrderLineItems :items="orderItems" class="lg:rounded" />
+      </VcCardWidget>
+
+      <!-- Items grouped by Vendor -->
+      <div
```
Comment (goldenmaya) [inline client-app/pages/account/order-details.vue:50]: wrap in the VcSectionWidget component

File: client-app/pages/account/order-details.vue [hunk 5]
```diff
@@ -1,93 +1,152 @@
...
+            <!-- Vendor -->
+            <VcVendor v-if="group.vendor" :vendor="group.vendor" class="pb-3"></VcVendor>
+
+            <OrderLineItems :items="group.items" class="lg:rounded" />
           </div>
+        </template>
+      </div>
+
+      <!-- Items not grouped by Vendor -->
+      <div v-else class="bg-white lg:mb-6 lg:rounded -mx-5 md:mx-0 lg:shadow-md-x lg:pt-5 lg:px-7">
```
Comment (goldenmaya) [inline client-app/pages/account/order-details.vue:68]: wrap in the VcSectionWidget component

File: client-app/pages/account/order-details.vue [hunk 6]
```diff
@@ -151,49 +137,21 @@
...
-
-            <p v-if="billingAddress?.email">
-              <span class="font-extrabold">
-                {{ $t("pages.account.order_details.billing_address_card.email_label") }}
-              </span>
-              {{ billingAddress?.email }}
-            </p>
+          <div class="flex flex-row items-center space-x-4">
+            <VcImage src="/static/images/checkout/invoice.svg" class="h-12 w-12" lazy />
+            <span class="overflow-x-hidden break-words">{{ order?.inPayments[0]?.paymentMethod?.typeName }}</span>
```
Comment (goldenmaya) [inline client-app/pages/account/order-details.vue:150]: remove overflow-x-hidden

File: client-app/ui-kit/components/molecules/vendor/vc-vendor.vue [hunk 7]
```diff
@@ -1,15 +1,20 @@
 <template>
   <div class="vc-vendor">
     <VcIcon name="vendor" size="sm" class="vc-vendor__icon" />
-    <span class="vc-vendor__title">{{ $t("common.labels.vendor") }}: {{ vendor.name }}</span>
+    <span class="vc-vendor__title">
+      {{ $t("common.labels.vendor") }}:
+      <span :class="['vc-vendor__name', vendor?.name ?? 'vc-vendor__name--not-available']">
```
Comment (artmakarov) [inline client-app/ui-kit/components/molecules/vendor/vc-vendor.vue:6]: `['vc-vendor__name', { 'vc-vendor__name--not-available' !vendor?.name } ]`

File: locales/en.json [hunk 8]
```diff
@@ -72,7 +76,10 @@
       "purchase_order_number": "Purchase order",
       "promotion_code": "Promotion code",
       "vendor": "Vendor",
-      "secure_checkout": "Secure checkout"
+      "secure_checkout": "Secure checkout",
+      "creates_label": "Creates",
```
Comment (artmakarov) [inline locales/en.json:80]: `"creates": "Creates",`

**Review by artmakarov — APPROVED** [review]

**Review by vkrashenko — APPROVED** [review]

**Pull request merged**